packages/go_router/example/lib/stateful_nested_navigation.dart

```diff
@@ -0,0 +1,292 @@
+              ),
+            ],
+          ),
+
+          /// The screen to display as the root in the second tab of the bottom
+          /// navigation bar.
+          GoRoute(
+            path: '/b',
+            builder: (BuildContext context, GoRouterState state) =>
+                const RootScreen(
+                    label: 'B',
+                    detailsPath: '/b/details/1',
+                    detailsPath2: '/b/details/2'),
+            routes: <RouteBase>[
+              /// The details screen to display stacked on navigator of the
+              /// second tab. This will cover screen B but not the application
+              /// shell (bottom navigation bar).
+              GoRoute(
+                path: 'details/:param',
+                builder: (BuildContext context, GoRouterState state) =>
+                    DetailsScreen(label: 'B', param: state.params['param']),
+              ),
+            ],
+          ),
+        ],
+      ),
+    ],
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      routerConfig: _router,
+    );
+  }
+}
+
+/// Representation of a tab item in a [ScaffoldWithNavBar]
+class ScaffoldWithNavBarTabItem extends BottomNavigationBarItem {
+  /// Constructs an [ScaffoldWithNavBarTabItem].
+  const ScaffoldWithNavBarTabItem(
+      {required this.navigationItem, required Widget icon, String? label})
+      : super(icon: icon, label: label);
+
+  /// The [StackedNavigationItem]
+  final StackedNavigationItem navigationItem;
+
+  /// Gets the associated navigator key
+  GlobalKey<NavigatorState> get navigatorKey => navigationItem.navigatorKey;
+}
+
+/// Builds the "shell" for the app by building a Scaffold with a
+/// BottomNavigationBar, where [child] is placed in the body of the Scaffold.
+class ScaffoldWithNavBar extends StatelessWidget {
+  /// Constructs an [ScaffoldWithNavBar].
+  const ScaffoldWithNavBar({
+    required this.currentIndex,
+    required this.itemsState,
+    required this.body,
+    required this.tabs,
+    Key? key,
+  }) : super(key: key ?? const ValueKey<String>('ScaffoldWithNavBar'));
+
+  /// Currently active tab index
+  final int currentIndex;
+
+  /// Route state
+  final List<StackedNavigationItemState> itemsState;
+
+  /// Body, i.e. the index stack
+  final Widget body;
+
+  /// The tabs
+  final List<ScaffoldWithNavBarTabItem> tabs;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: body,
+      bottomNavigationBar: BottomNavigationBar(
+        items: tabs,
+        currentIndex: currentIndex,
+        onTap: (int tappedIndex) =>
+            _onItemTapped(context, itemsState[tappedIndex]),
+      ),
+    );
+  }
+
+  void _onItemTapped(
+      BuildContext context, StackedNavigationItemState itemState) {
+    GoRouter.of(context).go(itemState.currentLocation);
```

Comment:
It will be better if `StackedNavigationItemState` can be access through context, we can consider add some an `of` method in `StackedNavigationShell` and expose the StackedNavigationItemState or some other more specific api in from the inherited widget to navigator between page.

Reply:
👍 Added static `of` method for this to StatefulShellRoute.